### PR TITLE
Add †Makefile† support

### PR DESCRIPTION
**Pull request by himanoa**
This PR add support make target list filter. 😃 
The implementation does't support `-f` option. 😢 

## 💯Demo 🖼 

makefile

### CODE REVIEW

**Comment by himanoa** [comment]
Should I add README's supported commands section?

File: tests/make.zunit [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env zunit
+
+@setup {
+    load ../fzf-zsh-completions.plugin.zsh
+    TEMP=$(mktemp -d)
+    pushd $TEMP
+}
+
+@test 'Testing completion: make **' {
+    MAKEFILE_CONTENT=$(cat <<< "
```
Comment (chitoku-k) [inline tests/make.zunit]: Can you just put this content out of the test file and move to `tests/_support` as `npm` does?

File: tests/make.zunit [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env zunit
+
+@setup {
+    load ../fzf-zsh-completions.plugin.zsh
+    TEMP=$(mktemp -d)
+    pushd $TEMP
+}
+
```
Comment (chitoku-k) [inline tests/make.zunit]: This test lacks of teardown section that wipes out the temporary directory.

File: src/make.zsh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env zsh
+
+
```
Comment (chitoku-k) [inline src/make.zsh]: Remove this line!

File: src/make.zsh [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+_fzf_complete_make() {
+    _fzf_complete "--ansi --tiebreak=index $fzf_options" $@ < <(grep -E '^[a-zA-Z_-]+:.*?$$' Makefile | sort | awk -F ':' '{ print $1  }')
```
Comment (chitoku-k) [inline src/make.zsh]: Why does it sort the result? It looks it's more obvious if they appear in the declaration order in Makefile though I'm not sure about this😉

P.S.
Nitpicking formatting issue:
`{ print $1 }`

File: tests/make.zunit [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    echo $MAKEFILE_CONTENT > Makefile
+    _fzf_complete() {
+        assert $# equals 2
+				
+        assert $2 same_as 'make '
+        run cat
+        assert ${lines[1]} same_as 'akane'
+        assert ${lines[2]} same_as 'aoi'
+      }
```
Comment (chitoku-k) [inline tests/make.zunit]: Indentation issue. I would suggest you edit this file with `filetype=zsh`

**Comment by chitoku-k** [comment]
README is going to get refactored in a few days. No worries!

File: tests/make.zunit (hunk 1 above)
Comment (himanoa) [reply tests/make.zunit]: OK. It will fix it.

File: tests/make.zunit (hunk 2 above)
Comment (himanoa) [reply tests/make.zunit]: 🙆

File: src/make.zsh (hunk 4 above)
Comment (himanoa) [reply src/make.zsh]: I have no particular opinion for sort. I'll fix it.

**Comment by himanoa** [comment]
@chitoku-k This should the trick

File: src/make.zsh [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env zsh
+
+_fzf_complete_make() {
+    _fzf_complete "--ansi --tiebreak=index $fzf_options" $@ < <(grep -E '^[a-zA-Z_-]+:.*?$$' Makefile | uniq | awk -F ':' '{ print $1 }')
```
Comment (chitoku-k) [inline src/make.zsh]: Add `2> /dev/null` so nothing is printed when this is run outside of repository.

File: tests/make.zunit [hunk 7]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env zunit
+
+@setup {
+    load ../fzf-zsh-completions.plugin.zsh
+    TEMP=$(mktemp -d)
```
Comment (chitoku-k) [inline tests/make.zunit]: My bad... I don't think we need TEMP for this case. Just `pushd` to `tests/_support/make`.

File: src/make.zsh (hunk 6 above)
Comment (chitoku-k) [reply src/make.zsh]: fzf_options is undefined in this path. Can you just remove that?

File: tests/make.zunit (hunk 7 above)
Comment (himanoa) [reply tests/make.zunit]: Ok.

**Pull request merged**